### ISSUE TITLE
db_filename is relative to the directory where the config file resides

### ISSUE THREAD

**Issue by fgsch** [sopel-irc/sopel#1047]
This should be relative to homedir. The way is currently handle makes no sense IMO.

**Comment by maxpowa** [comment]
If you want relative to homedir, use ~. This behavior is relatively common.

**Comment by fgsch** [comment]
I appreciate you can override it and that's what I did. However I believe this should be the default behaviour.

**Comment by dgw** [comment]
`~` is a de facto standard in the *nix world that represents the home directory of the user running a process. While one could argue that `.` as a reference to the current directory is equally common, and thus a path beginning `./` is clearly meant to be relative, I believe it is more common to default to a path relative to the current file. For example, configurations in nginx and apache that include other files look in the current directory first, too, not the webserver's home directory.

Long story short, it's a common convention that file references in configuration files are relative to the directory of the file they're in, unless otherwise specified by a leading `~` or `/`. For that reason, I doubt this will be changed in sopel.

**Comment by fgsch** [comment]
@dgw I appreciate that and that is indeed the way nginx and other software handles it.
In this particular case I don't think it's correct though.
To start, the `homedir` options reads:

```
homedir
The directory in which various files are stored at runtime.

By default, this is the same directory as the config. It can not be changed at runtime.
```

So by default, if `homedir` is not defined, the directory where the configuration resides it's used.
If `homedir` is defined, however, I'd expect it to take over since `homedir` is ***the directory in which various files are stored at runtime***. `db_filename` falls into this category AFAICT.

In fact, if you set `homedir` the only file stored outside `homedir` is the database.

If we take a look at `db_filename`:

```
db_filename
The filename for Sopel’s database.
```

There is no mention about location or being special wrt homedir.

As such I think this is incorrect. It doesn't follow how other files are stored. Having to define both _homedir_ and _db_filename_ is IMO a bug.

**Comment by maxpowa** [comment]
Oh, you're referring to the config setting `homedir`, not the concept of a user's home directory. I agree, `db_filename` should be relative to that setting, and relative to the config file if unset. Since this would change some existing configurations, it'd probably be best to have input from @embolalia too.

**Comment by fgsch** [comment]
Yes, `homedir`. I didn't mention the user's homedir.
It's better to not [assume](http://uncyclopedia.wikia.com/wiki/Assume) :smile:

**Comment by dgw** [comment]
It's also better to be specific. The initial exchange really did sound like you meant the user's homedir, especially after Max pointed out the use of `~` and you said that's what should be default behavior.

Communicating is hard sometimes. But yes, now that I know that you mean `homedir` and not `~` I agree, too. :smile:
